pkg/generate/app/cmd/resolve.go

```diff
@@ -108,7 +106,10 @@ func Resolve(r *Resolvers, c *ComponentInputs, g *GenerationInputs) (*ResolvedCo
 	for _, repo := range repositories {
 		repo.SetContextDir(g.ContextDir)
 	}
-
+	// Environment File where the parameters are stored
+	if len(g.EnvironmentFile) != 0 && !app.IsPossibleEnvironmentFile(g.EnvironmentFile) {
+		return nil, fmt.Errorf("when --env-file is specified you must provide a valid env file location")
+	}
 	if len(g.Strategy) != 0 && len(repositories) == 0 && !g.BinaryBuild {
 		return nil, fmt.Errorf("when --strategy is specified you must provide at least one source code location")
 	}
```

Comment:
Oh great, thanks for correcting me. I just tried that without = & it works perfectly fine and for the second part, I get your point now, since we are trying to resolve a component type called repository, we have it here- makes sense.